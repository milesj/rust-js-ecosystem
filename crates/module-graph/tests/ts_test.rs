mod utils;

use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::*;

mod ts {
    use super::*;

    #[test]
    fn export_star() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(sandbox.path(), "ts/export-star.ts"));
    }

    #[test]
    fn export_named() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-named.ts"
        ));
    }

    #[test]
    fn export_default_function() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-func.ts"
        ));
        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-anon-func.ts"
        ));
    }

    #[test]
    fn export_default_class() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-class.ts"
        ));
        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-anon-class.ts"
        ));
    }

    #[test]
    fn export_default_reference() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-ref.ts"
        ));
    }

    #[test]
    fn export_default_interface() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-interface.ts"
        ));
    }

    // This doesn't seem to generate exports,
    // even though oxc is a valid enum type!
    #[test]
    fn export_default_enum() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "ts/export-def-enum.ts"
        ));
    }
}
