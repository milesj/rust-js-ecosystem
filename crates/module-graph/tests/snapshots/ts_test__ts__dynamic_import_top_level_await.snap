---
source: crates/module-graph/tests/ts_test.rs
expression: "generate_graph_for_file(sandbox.path(), \"ts/dyn-import-tla.ts\")"
---
{
    2: Module {
        exports: [
            Export {
                kind: Modern,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 0,
                        end: 26,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "number",
                    },
                ],
                type_only: false,
            },
            Export {
                kind: Modern,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 27,
                        end: 55,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "string",
                    },
                ],
                type_only: false,
            },
            Export {
                kind: Modern,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 56,
                        end: 83,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: ValueType,
                        symbol_id: None,
                        name: "Type",
                    },
                ],
                type_only: true,
            },
        ],
        fragment: None,
        id: 2,
        imports: [],
        package_name: None,
        path: "/private/root/ts/shared.ts",
        query: None,
        source: JavaScriptModule {
            package_type: Unknown,
            source: "export const number = 123;\nexport const string = \"abc\";\nexport type Type = boolean;\n",
            source_type: SourceType {
                language: TypeScript {
                    is_definition_file: false,
                },
                module_kind: Module,
                variant: Standard,
                always_strict: false,
            },
            stats: JavaScriptStats {
                dynamic_import_count: 0,
                export_statements: 3,
                exports_default: false,
                import_statements: 0,
                other_statements: 0,
                require_count: 0,
            },
        },
    },
    1: Module {
        exports: [],
        fragment: None,
        id: 1,
        imports: [
            Import {
                kind: AsyncDynamic,
                module_id: 2,
                source_request: "./shared",
                span: Span {
                    start: 34,
                    end: 52,
                },
                symbols: [],
                type_only: false,
            },
            Import {
                kind: AsyncDynamic,
                module_id: 2,
                source_request: "./shared",
                span: Span {
                    start: 83,
                    end: 101,
                },
                symbols: [
                    ImportedSymbol {
                        kind: Namespace,
                        source_name: None,
                        symbol_id: None,
                        name: "ns",
                    },
                ],
                type_only: false,
            },
            Import {
                kind: AsyncDynamic,
                module_id: 2,
                source_request: "./shared",
                span: Span {
                    start: 146,
                    end: 164,
                },
                symbols: [
                    ImportedSymbol {
                        kind: Value,
                        source_name: None,
                        symbol_id: None,
                        name: "number",
                    },
                    ImportedSymbol {
                        kind: Value,
                        source_name: None,
                        symbol_id: None,
                        name: "string",
                    },
                ],
                type_only: false,
            },
            Import {
                kind: AsyncDynamic,
                module_id: 2,
                source_request: "./shared",
                span: Span {
                    start: 211,
                    end: 229,
                },
                symbols: [
                    ImportedSymbol {
                        kind: Value,
                        source_name: Some(
                            "number",
                        ),
                        symbol_id: None,
                        name: "renumber",
                    },
                ],
                type_only: false,
            },
        ],
        package_name: None,
        path: "/root/ts/dyn-import-tla.ts",
        query: None,
        source: JavaScriptModule {
            package_type: Unknown,
            source: "export {};\n\n// Side-effects\nawait import(\"./shared\");\n\n// Default\nconst ns = await import(\"./shared\");\n\n// Named\nconst { number, string } = await import(\"./shared\");\n\n// Renamed\nlet { number: renumber } = await import(\"./shared\");\n",
            source_type: SourceType {
                language: TypeScript {
                    is_definition_file: false,
                },
                module_kind: Module,
                variant: Standard,
                always_strict: false,
            },
            stats: JavaScriptStats {
                dynamic_import_count: 4,
                export_statements: 1,
                exports_default: false,
                import_statements: 0,
                other_statements: 3,
                require_count: 0,
            },
        },
    },
}
